clientupdate/clientupdate.go

```diff
@@ -1148,6 +1145,64 @@ func (up *Updater) updateQNAP() (err error) {
 	return nil
 }
 
+func (up *Updater) updateUnraid() (err error) {
+	if up.Version != "" {
+		return errors.New("installing a specific version on Unraid is not supported")
+	}
+	if err := requireRoot(); err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf(`%w; you can try updating using "plugin check tailscale.plg && plugin update tailscale.plg"`, err)
+		}
+	}()
+
+	// We need to run `plugin check` for the latest tailscale.plg to get
+	// downloaded. Unfortunately, the output of this command does not contain
+	// the latest tailscale version available. So we'll parse the downloaded
+	// tailscale.plg file manually below.
+	out, err := exec.Command("plugin", "check", "tailscale.plg").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to check if Tailscale plugin is upgradable: %w, output: %q", err, out)
+	}
+
+	// Note: 'plugin check' downloads plugins to /tmp/plugins.
+	// The installed .plg files are in /boot/config/plugins/, but the pending
+	// ones are in /tmp/plugins. We should parse the pending file downloaded by
+	// 'plugin check'.
+	latest, err := parseUnraidPluginVersion("/tmp/plugins/tailscale.plg")
+	if err != nil {
+		return fmt.Errorf("failed to find latest Tailscale version in /boot/config/plugins/tailscale.plg: %w", err)
+	}
+	if !up.confirm(latest) {
+		return nil
+	}
+
+	up.Logf("c2n: running 'plugin update tailscale.plg'")
+	cmd := exec.Command("plugin", "update", "tailscale.plg")
+	cmd.Stdout = up.Stdout
+	cmd.Stderr = up.Stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed tailscale plugin update: %w", err)
+	}
+	return nil
+}
+
+func parseUnraidPluginVersion(plgPath string) (string, error) {
```

Comment:
I could add the Tailscale version as an attribute to the <PLUGIN> tag (e.g., as tailscaleVersion="1.54.0"). That's probably a little nicer than having to regex it out of the file system :)

Reply:
I'd like to avoid having a hard dependency on git repo URLs. For example, you may chance your github username, move the repo under an org or transfer ownership to a new maintainer. All of these would likely break tag-based lookups.
Regexp-parsing the plugin file is gross, but at least it's not tied to where the plugin was installed from.

I don't know if it's possible, but making release versions match Tailscale versions (instead of being [dates](https://github.com/dkaser/unraid-tailscale/blob/49add5ead59357eae0c9b8c7e895a400d7a6ab27/plugin/tailscale.plg#L7)) would let me parse the output of `plugin check tailscale.plg` instead